templates/CRM/MembershipExtras/Page/NextPeriodTab.tpl

```diff
@@ -1,22 +1,92 @@
 <script>
 var recurringContributionID = {$recurringContributionID};
+var financialTypes = JSON.parse('{$financialTypes|@json_encode}');
+var disableClicks = false;
 
 {literal}
   CRM.$(function () {
+
     CRM.$('.remove-next-period-line-button').each(function () {
       CRM.$(this).click(function (e) {
         e.preventDefault();
+        
+        if(!disableClicks) {
+          var itemData = CRM.$(this).closest('tr').data('item-data');
+          showNextPeriodLineItemRemovalConfirmation(itemData);
 
-        var itemData = CRM.$(this).closest('tr').data('item-data');
-        showNextPeriodLineItemRemovalConfirmation(itemData);
-
-        CRM.$('#periodsContainer').on('crmLoad', function(event, data) {
-          CRM.$('#tab_next a').click();
-        });
+          CRM.$('#periodsContainer').on('crmLoad', function(event, data) {
+            CRM.$('#tab_next a').click();
+          });
+        }
       });
     });
+
+    CRM.$('#next_buttons #addOtherAmount').on('click', function(e) {
+      e.preventDefault();
+
+      if (!disableClicks) {
+        CRM.$('#addLineItemRow').show();
+        CRM.$('#periodsContainer').find('*').not(CRM.$('#addLineItemRow').find('*')).css('cursor', 'not-allowed');
+        CRM.$('#addLineItemRow').css('cursor', 'auto');
+        CRM.$('#addLineItemRow').css('opacity', 1);
+
+        disableClicks = true;
+      }
+    });
+
+    CRM.$('.cancel-add-next-period-line-button').on('click', function(e) {
+      e.preventDefault();
+
+      CRM.$('#addLineItemRow').hide();
+      CRM.$('#periodsContainer').find('*').css('cursor', 'auto');
+    });
+
+    CRM.$('#financialType').on('change', function() {
+      var selectedId = CRM.$(this).val();
+      var financialType = getFinancialType(selectedId);
+
+      if (!financialType) {
+        throw new Error('Invalid financial type id passed');
+      }
+      
+      CRM.$('#financialTypeTaxRate').text(financialType.tax_rate || 'N/A');
+    });
+
+    CRM.$('.confirm-add-next-period-line-button').on('click', function(e) {
+      e.preventDefault();
+      
+      var label = CRM.$('#item').val(),
+          amount = CRM.$('#amount').val(),
+          financial_type_id = CRM.$('#financialType').val();
+
+      if (!label.length) {
+        CRM.alert('Item label is required', null, 'error');
+        return;
+      }
+
+      if (!amount.length) {
+        CRM.alert('Item amount is required', null, 'error');
+        return;
+      } else {
+        try {
+          amount = parseInt(amount);
```

Comment:
Amounts can be floating point values, so you should probably use `parseFloat` instead.

Reply:
Maybe `isNan()` would be better?